src/OpenTelemetry.Exporter.Prometheus/Implementation/PrometheusMetricBuilder.cs

```diff
@@ -28,6 +28,8 @@ internal sealed class PrometheusMetricBuilder
     {
         public const string ContentType = "text/plain; version = 0.0.4";
 
+        internal static Func<DateTimeOffset> GetUtcNowDateTimeOffset = () => DateTimeOffset.UtcNow;
```

Comment:
Do we really need/want this to be static?

Reply:
Makes the tests much easier to write. What's your issue with it? Note the value isn't static, it is wrapping `DateTimeOffset.UtcNow` access into a delegate which the tests can swap out to make things deterministic.

Reply:
My worry is that if folks use this in multiple test cases and they run the tests simultaneously, it'll cause race condition (with different unit tests expecting different angel clocks)

Reply:
I was thinking we would just control the parallelism of the WriteMetricsCollection tests, but it was pretty easy to make it instance-based so I changed it.